mne/simulation/source.py

```diff
@@ -312,3 +317,204 @@ def simulate_stc(src, labels, stc_data, tmin, tstep, value_fun=None,
+
+        Returns a SourceEstimate object constructed according to the simulation
+        parameters which should be added through function add_data. If both
+        start_sample and stop_sample are not specified, the entire duration is
+        used.
+
+        Parameters
+        ----------
+        start_sample : int
+            First sample in chunk. Default is 0.
+        stop_sample : int | None
+            The stop sample of the returned stc. This sample is not part of the
+            output to follow slicing semantics. If None, then all samples past
+            start_sample is returned.
+
+        Returns
+        -------
+        stc : SourceEstimate object
+            The generated source time courses.
+        """
+        if len(self._labels) == 0:
+            raise ValueError('No simulation parameters were found. Please use '
+                             'function add_data to add simulation parameters.')
+        if stop_sample is None:
+            stop_sample = self.n_times
+
+        n_samples = stop_sample - start_sample
+
+        # Initialize the stc_data array
+        stc_data = np.zeros((len(self._labels), n_samples))
+
+        # Select only the indices that have events in the time chunk
+        ind = np.where(np.logical_and(self._last_samples >= start_sample,
+                                      self._events[:, 0] < stop_sample))[0]
+
+        # Loop only over the items that are in the time chunk
+        subset_waveforms = [self._waveforms[i] for i in ind]
+        for i, (waveform, event) in enumerate(zip(subset_waveforms,
+                                                  self._events[ind])):
+            # We retrieve the first and last sample of each waveform
+            # According to the corresponding event
+            wf_start = event[0]
+            wf_stop = self._last_samples[ind[i]]
+
+            # Recover the indices of the event that should be in the chunk
+            waveform_ind = np.in1d(np.arange(wf_start, wf_stop),
+                                   np.arange(start_sample, stop_sample))
+
+            # Recover the indices that correspond to the overlap
+            stc_ind = np.in1d(np.arange(start_sample, stop_sample),
+                              np.arange(wf_start, wf_stop))
+
+            # add the resulting waveform chunk to the corresponding label
+            stc_data[ind[i]][stc_ind] += waveform[waveform_ind]
+
+        stc = simulate_stc(self._src, self._labels, stc_data,
+                           start_sample * self._tstep, self._tstep,
+                           allow_overlap=True)
+
+        return stc
+
+    def __iter__(self):
+        # Arbitrary chunk size, can be modified later to something else
+        # Loop over chunks of 1 second - or, maximum sample size.
+        # Can be modified to a different value.
+        n_times = self.n_times
+        for start_sample in range(0, n_times, self._chk_duration):
```

Comment:
This `range` is not correct -- if you have `n_times=10` and a chunk duration of 6 this will give you a single chunk that is 6 samples long. You need to at the end yield a STC with however samples you have remaining (if any). There should be a unit test for this

Reply:
I did a quick test and it behaves correctly. With your example, you will get two start samples, 0 and 6. The number of samples of the stc is computed in the next line. I will still add a test for this.

Reply:
ahh yes I think you're right

I think it's good to go once the (additional) merge conflict is fixed -- I'll fix it in a few minutes unless you're on it already